plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/ValidationUtils.java

```diff
@@ -41,26 +41,25 @@
    * character offsets.
    */
   public static Map<BannedElement, Integer> getOffsetMap(byte[] bytes,
-      SaxParserResults parserResults) {
-    Queue<BannedElement> blacklist = parserResults.getBlacklist();
+      ArrayList<BannedElement> blacklist, String encoding) {
     Map<BannedElement, Integer> bannedElementOffsetMap = new HashMap<>();
-    ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
-    try (BufferedReader reader =
-        new BufferedReader(new InputStreamReader(bais, parserResults.getEncoding()))) {
-      int currentLine = 1;
-      int charOffset = 0;
-      while (!blacklist.isEmpty()) {
-        BannedElement element = blacklist.poll();
-        while (element.getStart().getLineNumber() > currentLine) {
-          String line = reader.readLine();
-          charOffset += line.length() + 1;
-          currentLine++;
-        }
-        int start = charOffset + element.getStart().getColumnNumber() - 1;
-        bannedElementOffsetMap.put(element, start);
+    for (BannedElement element : blacklist) {
+      ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+      try (BufferedReader reader =
+          new BufferedReader(new InputStreamReader(bais, encoding))) {
+          int currentLine = 1;
+          int charOffset = 0;
+          while (element.getStart().getLineNumber() > currentLine) {
+            String line = reader.readLine();
+            charOffset += line.length() + 1;
+            currentLine++;
```

Comment:
So in the end, `currentLine` is always `element.getStart().getLineNumber()`? Why don't you assign it directly?